apps/betterangels-backend/notes/tests/test_queries.py

```diff
@@ -191,6 +195,45 @@ def test_notes_paginated_query(self) -> None:
         note_differences = DeepDiff(self.note, notes[0], ignore_order=True)
         self.assertFalse(note_differences)
 
+    def test_interaction_authors(self) -> None:
+        query = """
+            query ViewInteractionAuthors {
+                interactionAuthors {
+                    totalCount
+                    results {
+                        id
+                        firstName
+                        lastName
+                        middleName
+                    }
+                }
+            }
+        """
+
+        test_user_map = {
+            "test_new_interaction_author": baker.make(User, username=f"test_new_interaction_author_{uuid.uuid4()}")
+        }
+
+        test_new_interaction_author = test_user_map["test_new_interaction_author"]
+        test_new_interaction_author.first_name = "Wanda"
+        test_new_interaction_author.last_name = "Maximoff"
+        test_new_interaction_author.middle_name = "J."
+        test_new_interaction_author.save()
+
+        caseworker_permission_group_template = PermissionGroupTemplate.objects.get(name="Caseworker")
+        perm_group = permission_group_recipe.make(template=caseworker_permission_group_template)
+        perm_group.organization.add_user(test_new_interaction_author)
+
+        response = self.execute_graphql(query)
+
+        self.assertFalse(response["data"] is None, f"Data is None: {response}")
+
+        firstNames = [result["firstName"] for result in response["data"]["interactionAuthors"]["results"]]
+
+        self.assertTrue("Wanda" in firstNames)
+
+        self.assertFalse(self.client_user_1.first_name in firstNames)
```

Comment:
`next` will return the first user that matches the condition (`u["id"] == str(case_manager.pk)`), and avoid iterating over the entire list. This approach also gets back the entire user dict for the target user, rather than just a list of names. This way, you can assert that all the fields returned as expected

```suggestion
        results = response["data"]["interactionAuthors"]["results"]
        returned_author: dict = next((u for u in results if u["id"] == str(case_manager.pk)))

        self.assertEqual(returned_author["firstName"], "Wanda")
        self.assertEqual(returned_author["lastName"], "Maximoff")
        self.assertEqual(returned_author["middleName"], "J.")
```